site/src/pages/ManagementSettingsPage/Sidebar.tsx

```diff
@@ -14,27 +17,45 @@ import { SidebarView } from "./SidebarView";
  * DeploySettingsPage/Sidebar instead.
  */
 export const Sidebar: FC = () => {
+  const location = useLocation();
   const { permissions } = useAuthenticated();
   const { organizations } = useOrganizationSettings();
   const { organization: organizationName } = useParams() as {
     organization?: string;
   };
 
-  // If there is no organization name, the settings page will load, and it will
-  // redirect to the default organization, so eventually there will always be an
-  // organization name.
-  const activeOrganization = organizations?.find(
-    (o) => o.name === organizationName,
-  );
-  const activeOrgPermissionsQuery = useQuery(
-    organizationPermissions(activeOrganization?.id),
+  const orgPermissionsQuery = useQuery(
+    organizationsPermissions(organizations?.map((o) => o.id)),
   );
 
+  // Sometimes a user can read an organization but cannot actually do anything
+  // with it.  For now, these are filtered out so you only see organizations you
+  // can manage in some way.
+  const editableOrgs = organizations
+    ?.map((org) => {
+      return {
+        ...org,
+        permissions: orgPermissionsQuery.data?.[org.id],
+      };
+    })
+    // TypeScript is not able to infer whether permissions are defined from the
+    // canEditOrganization call, so although redundant this helps figure it out.
+    .filter(
+      (org): org is OrganizationWithPermissions =>
+        org.permissions !== undefined,
+    )
+    .filter((org) => {
+      return canEditOrganization(org.permissions);
+    });
```

Comment:
could you just do...

```suggestion
    .filter(
      (org) => org.permissions && canEditOrganization(org.permissions);
    });
```

...instead? or is it still not happy with that? I try to avoid consecutive `.filter` calls

Reply:
As fortune would have it, I did indeed try exactly this first but sadly no dice.  It really seems like it should have worked though.

Tried it again now just in case I had messed it up the first time, but sadly:

```
Type '{ permissions: AuthorizationResponse | undefined; description: string; created_at: string; updated_at: string; is_default: boolean; id: string; name: string; display_name: string; icon: string; }[] | undefined' is not assignable to type 'OrganizationWithPermissions[] | undefined'.
```

Reply:
Oh wait duh I can just move `is OrganizationWithPermissions` to the second filter and get rid of the first.